### PR TITLE
TranslationProject initialisation cleanup

### PR DESCRIPTION
**Pull request by phlax**
Touch #4215

### CODE REVIEW

**Comment by phlax** [comment]
@ta2-1 if you get a chance can you take a look at this  - tia

**Comment by phlax** [comment]
@julen i reckon this is pretty much cleaned up - if you or ta2-1 could take a look that would be really appreciated

File: docs/server/installation.rst [hunk 1]
```diff
@@ -172,23 +172,12 @@ the database and populate it with initial data. This is done by executing the
...
-
-.. _installation#refreshing-stats:
-
-Refreshing stats
-----------------
-
-On first installation you will need to generate the statistics from your
-database. You will need to have an :ref:`RQ worker running 
+You will need to have an :ref:`RQ worker running 
 <installation#running-rqworker>` to complete this.
```
Comment (unho) [inline docs/server/installation.rst]: This warning should be before the snippet with the commands to run.

File: docs/server/installation.rst [hunk 2]
```diff
@@ -172,23 +172,12 @@ the database and populate it with initial data. This is done by executing the
...
 
-.. code-block:: bash
-
-   (env) $ pootle refresh_stats
-
 This command will dispatch jobs to the RQ worker and may take some time.
```
Comment (unho) [inline docs/server/installation.rst]: Which command? `migrate`? `initdb`?

File: tests/fixtures/models/language.py [hunk 3]
```diff
@@ -27,13 +27,14 @@ def _require_language(code, fullname, plurals=2, plural_equation='(n != 1)'):
         'nplurals': plurals,
         'pluralequation': plural_equation,
     }
+
```
Comment (unho) [inline tests/fixtures/models/language.py]: Can we just avoid adding these blank lines? Or at least put them on a separate commit that can be directly landed as this change is not directly related with any change in the surrounding code. Note that I am ok to keep this to allow this PR to be merged asap, so consider this as a reminder for the future.

File: tests/fixtures/models/language.py [hunk 4]
```diff
@@ -84,3 +85,18 @@ def russian(english):
...
+@pytest.fixture
+def klingon(english):
+    """Require the Klingon language."""
+    return _require_language('kl', 'Klingon')
+
+
+@pytest.fixture
+def klingon_vpw(english):
+    """Require the Klingon language (VPW dialect)."""
+    return _require_language('kl_VPW', 'Klingon vegan peace warriors')
```
Comment (unho) [inline tests/fixtures/models/language.py]: You definitely need to do a Star Trek marathon. Perhaps twice :)

File: tests/models/revision.py [hunk 5]
```diff
@@ -25,7 +25,6 @@ def test_max_revision(af_tutorial_po):
     initial_max_revision = Unit.max_revision()
     initial_revision = Revision.get()
     assert initial_max_revision == initial_revision
-    assert initial_max_revision == 1
```
Comment (unho) [inline tests/models/revision.py]: Is this change related to the rest of the changes on the PR?

Comment (phlax) [inline tests/models/revision.py]: yes - the test fails without removing this line

File: tests/models/translationproject.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+from pootle_language.models import Language
+from pootle_project.models import Project
+from pootle_translationproject.models import TranslationProject
+
+
+@pytest.mark.django_db
+def test_tp_create_fail(tutorial, english):
+
+    # trying to create a TP with no Project raises a RelatedObjectDoesNotExist
+    # which can be caught with Project.DoesNotExist
```
Comment (unho) [inline tests/models/translationproject.py]: Nit: I don't particularly like texts starting with lowercase and without ending period. Every time I see one of these I have doubts about if part of the comment has been removed at the beginning or at the end, or in other words if comment is incomplete.

File: docs/server/installation.rst (hunk 1 above)
Comment (unho) [inline docs/server/installation.rst]: This is because some readers that are in a hurry might just copy and paste the commands, run them and then finish reading the instructions and notice that they ran something they actually shouldn't.

File: tests/fixtures/models/language.py [hunk 7]
```diff
@@ -27,13 +27,14 @@ def _require_language(code, fullname, plurals=2, plural_equation='(n != 1)'):
...
     }
+
     language, created = Language.objects.get_or_create(**criteria)
 
     return language
 
 
 @pytest.fixture
-def english(root):
+def english(root, system):
```
Comment (unho) [inline tests/fixtures/models/language.py]: This looks like some bugfix that should go on its own commit, isn't it?

File: tests/fixtures/models/language.py [hunk 8]
```diff
@@ -84,3 +85,18 @@ def russian(english):
 def fish(english):
     """Require the Fish language ><(((º>"""
     return _require_language(code='fish', fullname='Fish')
+
+
+# due to issues with tests not having a clean slate ie:
+# (https://github.com/translate/pootle/issues/3898)
+# please do not use the klingon fixtures 8)
```
Comment (unho) [inline tests/fixtures/models/language.py]: Which is the point on adding languages that cannot be used?

Comment (phlax) [inline tests/fixtures/models/language.py]: they are used

File: tests/fixtures/models/language.py (hunk 7 above)
Comment (phlax) [inline tests/fixtures/models/language.py]: no - its not - its never been necessary before

File: tests/models/translationproject.py [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+
+def _test_tp_stores_match(tp1, tp2):
+    # for testing tp creation from templates
+    assert tp1.stores.count() == tp2.stores.count()
+    tp1_stores = tp1.stores.all()
+    tp2_stores = tp2.stores.all()
+
+    for i, store1 in enumerate(tp1_stores):
+        store2 = tp2_stores[i]
```
Comment (unho) [inline tests/models/translationproject.py]: Perhaps simpler:

```python
    for store1, store2 in zip(tp1_stores, tp2_stores):
        assert store1.units == store2.units
```

File: tests/models/translationproject.py [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        assert store1.units == store2.units
```
Comment (unho) [inline tests/models/translationproject.py]: Perhaps simpler:

```python
    for store1, store2 in zip(tp1_stores, tp2_stores):
        assert store1.units == store2.units
```

Comment (phlax) [inline tests/models/translationproject.py]: nice - but not important right now - this function is not currently used

File: tests/models/translationproject.py [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+    assert tp.stores.count() == template_tp.stores.count()
+    assert (
+        [(s, t)
+         for s, t
+         in template_tp.stores.first().units.values_list("source_f",
+                                                         "target_f")]
+        == [(s, t)
+            for s, t
+            in tp.stores.first().units.values_list("source_f",
+                                                   "target_f")])
```
Comment (unho) [inline tests/models/translationproject.py]: I bet we can make this a bit more unreadable by using Perl. What about using a helper function for getting the list for each of the TPs here?

**Comment by unho** [comment]
lgtm

File: tests/models/translationproject.py (hunk 11 above)
Comment (phlax) [inline tests/models/translationproject.py]: post 2.7.3 these tests will be tidied
